docs: add kagent kubernetes integration (#515)
agentgateway/website: assets/agw-docs/pages/integrations/kagent.md

[Kagent](https://github.com/kagent-dev/kagent) is a Kubernetes-native AI agent framework that brings autonomous agents to cloud-native environments. It leverages Kubernetes primitives for agent lifecycle management, scaling, and orchestration.

## What is Kagent?

Kagent provides a Kubernetes-native approach to running AI agents:

- **CRD-based Configuration** - Define agents as Kubernetes resources
- **Native Scaling** - Horizontal pod autoscaling for agent workloads
- **MCP Support** - Built-in Model Context Protocol for tool access
- **A2A Communication** - Agent-to-agent messaging via Kubernetes services
- **GitOps Ready** - Declarative agent definitions for Flux/ArgoCD

> ## Why Use agentgateway with Kagent?

Reviewer: Sentence case. Also one note throughout, `kagent` is spelled sentence case, so uppercase K at the start of a sentence, but lowercase k in the middle.

```suggestion
## Why use agentgateway with kagent?
```

Author: changed in d92ee9952ccc8ae12a4f87d324bf231024e2aba9

Kagent agents running in Kubernetes need enterprise governance:

| Kubernetes Challenge | agentgateway Solution |
|---------------------|----------------------|
| Multi-tenant clusters | Namespace-aware policies |
| Service-to-service auth | mTLS and JWT validation |
| Distributed tracing | OpenTelemetry integration |
| Cost allocation | Per-namespace token tracking |
| Compliance requirements | Centralized audit logging |

## Before you begin
{{< reuse "agw-docs/snippets/prereq.md" >}}

> 4. Follow the [Ollama](https://agentgateway.dev/docs/kubernetes/latest/llm/providers/ollama/) guide to install and setup Ollama

Reviewer: Shortcode for links in the same doc set.

And sentence-case.
```suggestion
4. Follow the [Ollama]({{< link-hextra path="llm/providers/ollama/">}}) guide to install and setup Ollama.
```

> ## Architecture

Reviewer: Add a short description:

```suggestion
## Architecture

This guide sets up kagent and agentgateway in a kind cluster, as shown in the following diagram.
```

```

> ┌───────────────────────────────────────────┐

Reviewer: We often use AI to turn diagrams into mermaid.js so that they render stylized. Not sure if you already tried that?

Author: cool tool, never used it before. changed in [604bee3](https://github.com/agentgateway/website/pull/515/commits/604bee3edd7ddbe06d61e57f121eb9c4c8da553b)

│              kind cluster                 │
│                                           │
│  ┌──────────┐    ┌─────────────────────┐  │   ┌────────────┐
│  │  kagent  │───▶│   agentgateway      │──│──▶│   Ollama   │
│  │  agent   │    │   (agentgateway-    │  │   │   (host)   │
│  │  pods    │    │    system ns)       │  │   └────────────┘
│  └──────────┘    │  - auth / authz     │  │
│                  │  - rate limiting    │  │
│                  │  - audit logging    │  │
│                  │  - observability    │  │
│                  └─────────────────────┘  │
└───────────────────────────────────────────┘
```

> ## Install Kagent

Reviewer: Maybe crosslink to the kagent docs.
```suggestion
## Install Kagent

Install kagent in your cluster. For more information, see the [kagent docs](https://kagent.dev/docs/kagent/introduction/installation).
```

1. Install Kagent CRDs.
   ```shell
   helm install kagent-crds oci://ghcr.io/kagent-dev/kagent/helm/kagent-crds \
       --namespace kagent \
       --create-namespace
   ```

2. Install Kagent.
   ```shell
   helm install kagent oci://ghcr.io/kagent-dev/kagent/helm/kagent \
     --namespace kagent \
     --create-namespace \
     --set providers.default=ollama \
     --set providers.ollama.baseUrl=http://agentgateway-proxy.agentgateway-system.svc.cluster.local/v1 \
     --set providers.ollama.apiKey=dummy
   ```

3. Verify everything is up and running.
   ```shell
   kubectl get pods -n kagent
   ```

>    Example Output:

Reviewer: sentence case
```suggestion
   Example output:
```

   ```

## Setup Kagent
1. Create a `ModelConfig` that points to Ollama.
   ```yaml
   kubectl apply -f- <<EOF
   apiVersion: kagent.dev/v1alpha2
   kind: ModelConfig
   metadata:
     name: llama3-model-config
     namespace: kagent
   spec:
     model: llama3
     provider: Ollama
     ollama:
       host: agentgateway-proxy.agentgateway-system.svc.cluster.local
   EOF
   ```

2. Verify that Kagent is accessible and correctly functioning.
   {{< tabs items="Cloud Provider LoadBalancer,Port-forward for local testing" tabTotal="2"  >}}
   {{% tab tabName="Cloud Provider LoadBalancer" %}}
   ```sh
   export INGRESS_GW_ADDRESS=$(kubectl get svc -n kagent kagent-ui -o jsonpath="{.spec.clusterIP}")
   echo $INGRESS_GW_ADDRESS  
   ```
   {{% /tab %}}
   {{% tab tabName="Port-forward for local testing"  %}}
   ```shell
   kubectl port-forward -n kagent service/kagent-ui 8082:8080
   ```
   {{% /tab %}}
   {{< /tabs >}}

3. Open the Kagent UI and try the default `k8s-agent` to confirm everything works end-to-end.
   {{< reuse-image src="img/kagent-default-k8s-agent.png" >}}
   {{< reuse-image-dark srcDark="img/kagent-default-k8s-agent.png" >}}

> ## Governance Capabilities

Reviewer: Add a shortdesc explaining the point of this section
```suggestion
## Governance Capabilities

Agentgateway provides policies that you can use to govern your kagent environment.
```

### Block requests with PII

> 1. Create an `AgentgatewayPolicy` resource to deny requests to the LLM provider that include PII, such as a `email` string in the request body on. See the full [guardrails docs](docs/kubernetes/latest/llm/guardrails/regex/#block-requests-with-pii) for more options.

Reviewer: ```suggestion
1. Create an `AgentgatewayPolicy` resource to deny requests to the LLM provider that include PII, such as a `email` string in the request body on. For more examples, see the [Guardrails docs]({{< link-hextra path="/llm/guardrails/regex/#block-requests-with-pii">}}).
```

   ```yaml
   kubectl apply -f - <<EOF
   apiVersion: agentgateway.dev/v1alpha1
   kind: AgentgatewayPolicy
   metadata:
     name: prompt-guard
     namespace: agentgateway-system
   spec:
     targetRefs:
     - group: gateway.networking.k8s.io
       kind: HTTPRoute
       name: ollama
     backend:
       ai:
         promptGuard:
           request:
           - response:
               message: "Rejected due to inappropriate content"
             regex:
               action: Reject
               matches:
               - "email"
   EOF
   ```

> 2. Verify the policy by sending a prompt that includes the word `email`. And you shpuld get an `403`

Reviewer: I think this means in kagent? also a typo.
```suggestion
2. Verify the policy by sending a prompt to your agent through the kagent UI that includes the word `email`. You get a `403` response.
```

   {{< reuse-image src="img/kagent-rejected-content.png" >}}
   {{< reuse-image-dark srcDark="img/kagent-rejected-content.png" >}}

## Best Practices

1. **Namespace Isolation** - Separate teams with namespace-aware policies
2. **Resource Quotas** - Set token budgets per namespace
3. **Audit Everything** - Log all LLM, MCP, and A2A traffic

> 4. **GitOps Policies** - Manage authorization policies declaratively

Reviewer: I am not sure if this section on best practices is needed, it does not seem very kagent specific to me.

Author: It was part of the old docs, but I have no hard feelings about that. It just highlights what is possible.

Author: removed in [f5fae4c](https://github.com/agentgateway/website/pull/515/commits/f5fae4cd821bd4951df85b890294097019d3114d)
